cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2019 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "snap-confine-invocation.h"
+
+#include <string.h>
+
+#include "../libsnap-confine-private/snap.h"
+#include "../libsnap-confine-private/utils.h"
+
+void sc_init_invocation(sc_invocation *inv, const struct sc_args *args, const char *snap_instance) {
```

Comment:
it's a bit unclear if this merit a new file instead of being a static helper in snap-confine.c,  also "invocation" as a concept doesn't tell anything much to people new to this so while understand the need for the .h file I'm sceptical it's useful to have a dedicated module about it,  the fallback code in the other PR put here doesn't help the case. Overall the snap-confine code is already a bit spread out with sometimes uneven layering depth.

Reply:
Merged into snap-confine.[ch] now